puppet/lib/puppet/indirector/facts/puppetdb.rb

```diff
@@ -26,7 +26,18 @@ def save(request)
         facts = request.instance.dup
         facts.values = facts.values.dup
         facts.values[:trusted] = get_trusted_info(request.node)
-        {
+
+        inventory = facts.values['_puppet_inventory_1']
+        package_inventory =
+          if inventory and inventory.respond_to?(:keys) and inventory['packages']
+            inventory['packages'].map { |p|
+              [p['package_name'], p['version'], p['provider']]
```

Comment:
We talked about this; the reason we have maps in the fact but tuples in the command is to keep the more 'public' part of the data (i.e. the data generated on the agent, and sent from  the agent to the master) self describing. We should still be able to split out the array values in the fact definition - I asked Scott Walker to do just that in a comment on the design doc.